Cl 2076 fe skip this step is shown when skipping is not possible verification

brentguf opened the pull request:
We were showing a 'skip this step' button when a user needed to verify after assigning an idea to their basket because we were opening the sign up/in modal (even if they were logged in already). In the sign up/in modal, verification can always be skipped.

This very likely also solves CL-2077. To be checked.

Besides that, I also simplified AssignBudgetControl a bit 🍝

brentguf committed:
ShieldIcon styles

brentguf committed:
Open verification modal when logged in and needing to verify

brentguf committed:
Remove wrapper component

brentguf committed:
Narrow participationContextId type

brentguf committed:
Move idea/participationContextId to function parameters instead of conditional checks

brentguf committed:
Move basket service calling to seperate function

brentguf committed:
Move getAddRemoveButtonMessage outside component

brentguf committed:
Clarify signed in vs. signed out

brentguf committed:
Reduce bloated number of variables

brentguf committed:
openSignUpInModal context values

brentguf committed:
Remove unnecessary intermediary function

brentguf commented on front/app/components/AssignBudgetControl/index.tsx:
```diff
@@ -98,184 +87,181 @@ const StyledPBExpenses = styled(PBExpenses)`
+      ? project
+      : latestRelevantIdeaPhase;
+    const participationContextType = isContinuousProject ? 'project' : 'phase';
+    const participationContextId = participationContext?.id || null;
```
Deleted the 'wrapper' component and brought data loading and var creation inside the component to simplify it.

brentguf commented on front/app/components/AssignBudgetControl/index.tsx:
```diff
@@ -98,184 +87,181 @@ const StyledPBExpenses = styled(PBExpenses)`
-        await timeout(200);
-        setProcessing(false);
-      };
+    const handleAddRemoveButtonClick =
```
The only thing handleAddRemoveButtonClick was doing was event.preventDefault and calling assignBudget, while assignBudget was dealing with opening verification/sign up modals. Opening these is what the handler now deals with. The actual assigning to the budget remained in assignBudget.

brentguf commented on front/app/components/AssignBudgetControl/index.tsx:
```diff
@@ -98,184 +87,181 @@ const StyledPBExpenses = styled(PBExpenses)`
         if (
+          // not signed up/in
           isNilOrError(authUser) ||
-          budgetingDisabledReason === 'not_verified'
```
This never evaluated to true. When nog signed up/in, the disabled reason was always 'not_signed_in'.

brentguf commented on front/app/components/AssignBudgetControl/index.tsx:
```diff
@@ -98,184 +87,181 @@ const StyledPBExpenses = styled(PBExpenses)`
+          budgetingDisabledReason === 'not_signed_in'
         ) {
           openSignUpInModal({
-            verification: budgetingDisabledReason === 'not_verified',
```
verification doesn't work here. For one because this disabled reason check is always false here, but when I set it manually to true, it didn't work either. I'll log this as an extra bug.

IvaKop replied:
Definitely should check this at some point but agree it's separate.

brentguf commented on front/app/components/AssignBudgetControl/index.tsx:
```diff
@@ -98,184 +87,181 @@ const StyledPBExpenses = styled(PBExpenses)`
     };
 
     if (
-      !isUndefined(authUser) &&
```
No idea why we have this check.

brentguf commented on front/app/components/AssignBudgetControl/index.tsx:
```diff
@@ -98,184 +87,181 @@ const StyledPBExpenses = styled(PBExpenses)`
       !isUndefined(basket) &&
-      idea.attributes.budget
+      idea.attributes.budget &&
+      participationContextId
```
In the wrapper component, we did this check to ensure participationContextId that was passed was a string. By doing this check here (and passing the string to functions, the other places where it was used), we have the same effect.

brentguf commented on front/app/components/AssignBudgetControl/index.tsx:
```diff
@@ -98,184 +87,181 @@ const StyledPBExpenses = styled(PBExpenses)`
-      const getAddRemoveButtonMessage = (view: TView) => {
-        switch (view) {
-          case 'ideaCard':
-            if (isInBasket) {
-              return messages.remove;
-            } else {
-              return messages.add;
-            }
-          case 'ideaPage':
-            if (isInBasket) {
-              return messages.removeFromMyBasket;
-            } else {
-              return messages.addToMyBasket;
-            }
-        }
-      };
-
-      if (isPBContext) {
+      if (
+        participationContext?.attributes?.participation_method === 'budgeting'
```
Removed some unnecessary variables, where the path to the value was descriptive enough. I also moved the getAddRemoveButtonMessage outside of the component to clean it up some more.

brentguf commented on front/app/containers/ProjectsShowPage/shared/pb/PBExpenses.tsx:
```diff
@@ -259,7 +259,7 @@ const TooltipContentText = styled.div`
 
 interface Props {
-  participationContextId: string | null;
+  participationContextId: string;
```
It's a string everywhere we pass `participationContextId` so better to narrow the type.

brentguf commented on front/app/modules/commercial/verification/citizen/components/VerificationMethods.tsx:
```diff
@@ -72,10 +72,10 @@ const StyledAvatar = styled(Avatar)`
 `;
 
 const ShieldIcon = styled(Icon)`
+  opacity: 0.5;
```
We had some shield icons that were off since the upgrade, I made them look alike.

brentguf commented on front/app/modules/commercial/verification/citizen/components/VerificationStatus.tsx:
```diff
@@ -46,7 +46,7 @@ const StyledAvatar = styled(Avatar)`
 
 const ShieldIcon = styled(Icon)`
-  fill: ${colors.white};
+  fill: ${colors.textSecondary};
```
The icon here wasn't visible (white on a white background).

IvaKop reviewed (approved):
Agree this is a sane approach towards the bug and love the cleanup you did 🤩 Hopefully this fixes both bugs indeed. The only thing seems to be there's some dead code now, we should remove it before merging.

brentguf committed:
Merge branch 'master' of https://github.com/CitizenLabDotCo/citizenlab into CL-2076-fe-skip-this-step-is-shown-when-skipping-is-not-possible-verification

brentguf committed:
Changelog

brentguf committed:
Remove dead code

brentguf committed:
Get rid of lodash

brentguf commented:
@IvaKop if you want to do another quick check: I removed lodash-es from AssignBudgetControl (saw that we were doing the isUndefined check of the basket also for no reason). And replaced `includes`.

I'm already checking, so merging without your extra check in 15 minutes or so.
